.github/workflows/build-test.yml

```diff
@@ -58,32 +50,29 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
-        with:
-          node-version: "21.4"
-          registry-url: "https://registry.npmjs.org"
 
-      - name: Get yarn cache directory path
-        id: yarn-cache-dir-path
-        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8.15
 
-      - uses: actions/cache@v3
-        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+      - uses: actions/setup-node@v4
         with:
-          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-yarn-
+          node-version: '21.4'
+          cache: 'pnpm'
 
-      - name: Install project dependencies
-        run: yarn --prefer-offline
-        id: install
+      - name: Install Dependencies
+        run: pnpm install
 
+      - name: Build the project
+        run: pnpm run build
+      
       - name: Run Prettier
-        run: yarn prettier
+        run: pnpm prettier
 
       - name: Run Linter
-        run: yarn lint
+        run: pnpm lint
 
-      - name: Run Tests
-        run: yarn test
+      - name: Cypress run
+        uses: cypress-io/github-action@v5
+        with:
+          start: pnpm start, pnpm test
```

Comment:
please check @0xArdy

Reply:
(idk much about cypress best practices into actions)

Reply:
yeap, that's what Vercel recommends to use with Cypress

> Cypress will simulate a user navigating your application, this requires your Next.js server to be running. We recommend running your tests against your production code to more closely resemble how your application will behave.

https://nextjs.org/docs/pages/building-your-application/testing/cypress#running-e2e-tests